corehq/apps/hqmedia/models.py

```diff
@@ -153,14 +153,9 @@ def attach_data(self, data, original_filename=None, username=None, attachment_id
 
         if not self.blobs or attachment_id not in self.blobs:
             if not getattr(self, '_id'):
-                # put attchment blows away existing data, so make sure an id has been
-                # assigned to this guy before we do it. this is the expected path
+                # put_attchment blows away existing data, so make sure an id has been assigned
+                # to this guy before we do it. This is the usual path; remote apps are the exception.
                 self.save()
-            else:
-                # this should only be files that had attachments deleted while the bug
-                # was in effect, so hopefully we will stop seeing it after a few days
-                soft_assert(notify_admins=True)(False, '[ICDS-291] someone is uploading a file that should '
```

Comment:
This is a really old soft assert to fix a really old bug that, as it says, should no longer be an issue. However, this code path is the normal path for remote linked apps, where the multimedia already has an id, assigned by the master server, before the file gets uploaded to the linked server.